lib/worker.go

```diff
@@ -106,6 +117,21 @@ func (w *DefaultBaseWorker) Workload() model.RescUnit {
 }
 
 func (w *DefaultBaseWorker) Init(ctx context.Context) error {
+	// TODO refine this part
+	poolCtx, cancelPool := context.WithCancel(context.TODO())
```

Comment:
Can we use `context.WithCancel(ctx)` here?

Reply:
Yes we can. But we need to clearly specify what context the runtime needs to call `Init` with. The context used to call `Init` cannot be a context with a timeout, or otherwise the background goroutines can be canceled prematurely.